[SPARK-25760][SQL] Set AddJarCommand return empty

wangyum opened the pull request:
## What changes were proposed in this pull request?

Only `AddJarCommand` return `0`, the user will be confused about what it means. This PR sets it to empty.

```sql
spark-sql> add jar /Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar;
ADD JAR /Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar
0
spark-sql>
```

## How was this patch tested?

manual tests
```sql
spark-sql> add jar /Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar;
ADD JAR /Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar
spark-sql> 
```

wangyum committed:
Fix AddJarCommand result

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/testing-k8s-prb-make-spark-distribution-unified/4034/
Test PASSed.

SparkQA commented:
**[Test build #97456 has started](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97456/testReport)** for PR 22747 at commit [`0a5b0dd`](https://github.com/apache/spark/commit/0a5b0ddd6b17381b5ff9d2f0b4c437bdddd6c6a6).

srowen commented:
I tend to agree, but I wonder if it returns a row with 0 to emulate something else like Hive?

wangyum commented:
Hive will not return 0:
```sql
hive> add jar /Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar;
Added [/Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar] to class path
Added resources: [/Users/yumwang/spark/sql/hive/src/test/resources/TestUDTF.jar]
hive> 
```

srowen commented:
Interesting, this was changed a long time ago, with the claim that it was needed to match Hive: https://github.com/apache/spark/pull/4586#discussion_r28394029 Maybe it's changed again? CC @adrian-wang

SparkQA commented:
**[Test build #97456 has finished](https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/97456/testReport)** for PR 22747 at commit [`0a5b0dd`](https://github.com/apache/spark/commit/0a5b0ddd6b17381b5ff9d2f0b4c437bdddd6c6a6).
 * This patch passes all tests.
 * This patch merges cleanly.
 * This patch adds no public classes.

AmplabJenkins commented:
Merged build finished. Test PASSed.

AmplabJenkins commented:
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
https://amplab.cs.berkeley.edu/jenkins//job/SparkPullRequestBuilder/97456/
Test PASSed.

HyukjinKwon reviewed (approved):
I checked the Hive side code. I think its return code (that indicates the command was successful or not) is misunderstood. Looks not returning rows in Hive side and the current fix looks coherent to me.

dongjoon-hyun reviewed (approved):
+1, LGTM, too.

dongjoon-hyun commented:
BTW, @wangyum . Please create a minor JIRA issue. This is a behavior change for users.

cc @gatorsmile

dongjoon-hyun commented:
Merged to master.

HyukjinKwon commented:
This looks also external changes to existing application users. Shall we update migration guide?

srowen commented:
It seems like a bug fix more than anything, and I assume we wouldn't document every single one, but don't object to mentioning it if that's more standard.

srowen commented:
https://github.com/apache/spark/pull/22826
